client/client-lib/src/lib.rs

```diff
@@ -593,8 +607,10 @@ impl Client<UserClientConfig> {
     /// Notify gateway that we've escrowed tokens they can claim by routing our payment and wait
     /// for them to do so
     pub async fn await_outgoing_contract_execution(&self, contract_id: ContractId) -> Result<()> {
-        let gateway = self.fetch_gateway().await?;
-
+        let gateway = match self.fetch_active_gateway().await {
+            Some(g) => g,
+            None => self.switch_active_gateway(None).await?,
+        };
```

Comment:
I wonder if this logic should just go into fetch active gateway? Currently there's some code duplication here.

Another option: what if we ensured that there always was an active GW as long as we know any GWs? This would require checking if we already have an active GW on writes to the GW list in the DB and marking one active in case there's none.

Reply:
Great plug on moving the activation logic into `fetch_active_gateway()`. Provided there is at least one registered gateway, I think this pattern naturally guarantees the client an active gateway when they need one.

Perf conscious clients can frontload call to `fetch_active_gateway()` or `switch_active_gateway(fetch_registered_gateways()[n])` early in their session and expect subsequent `fetch_active_gateway()` calls to be fast, or cache these gateways as desired. WYT?